Pre_processing_scripts/01_Sort_IT_Extracts.R

```diff
@@ -0,0 +1,129 @@
+rm(list = ls())
+devtools::load_all()
+
+it_extract_path <- file.path(
+  get_slf_dir(),
+  "IT_extracts"
+)
+it_extract_anon_path <- file.path(
+  it_extract_path,
+  "anon-chi-IT"
+)
+
+is_chi_in_file <- function(filename) {
+  data <- read.csv(filename, nrow = 1)
+  return(grepl("UPI", names(data)) %>% any())
+}
+
+# detect task number and tidy up parquet files from last request ---------
+# One can manually delete files and comment this part out.
+get_previous_it_task_num <- function() {
+  existing_parquet_list <- list.files(it_extract_anon_path)
+  ltcs_file <-
+    existing_parquet_list[stringr::str_detect(existing_parquet_list, "LTCs\\.parquet$")]
+  task_num <- stringr::str_extract(ltcs_file, "SCTASK\\d+")
+  if (length(task_num) == 1L) {
+    return(task_num)
+  } else {
+    cli::cli_alert_danger("Detected 0 or more than 1 task numbers! Please check!")
+    return(NULL)
+  }
+}
+
+get_current_it_task_num <- function() {
+  new_csv_files <- list.files(it_extract_path)
+  ltcs_file <-
+    new_csv_files[stringr::str_detect(new_csv_files, "LTCs\\.csv$")]
+  task_num <- stringr::str_extract(ltcs_file, "SCTASK\\d+")
+  if (length(task_num) == 1L) {
+    return(task_num)
+  } else {
+    cli::cli_abort("Detected 0 or more than 1 task numbers! Please check!")
+  }
+}
+
+previous_task_num <- get_previous_it_task_num()
+current_task_num <- get_current_it_task_num()
+
+existing_parquet_files <- list.files(it_extract_anon_path, full.names = TRUE)
+if (is.null(previous_task_num)) {
+  cli::cli_alert_info("No IT Extracts from last update to be removed.")
+} else {
+  file.remove(existing_parquet_files[grepl(previous_task_num, existing_parquet_files)])
+}
+
+
+# convert csv to anon parquet ---------------------------------------------
+csv_files <- list.files(it_extract_path,
+  pattern = "SCTASK[0-9]{7}_(PIS_20[0-9]{2}|Deaths|LTCs)\\.csv(\\.gz)?",
+  full.names = TRUE
+)
+
+convert_it_csv_to_parquet <- function(csv_file) {
+  parequet_file <- gsub("\\.csv(\\.gz)?$", ".parquet", csv_file)
+  data <- read_file(csv_file)
+  new_file <- file.path(
+    it_extract_anon_path,
+    paste0("anon-", basename(parequet_file))
+  )
+  if (is_chi_in_file(csv_file)) {
+    data <- data %>%
+      dplyr::rename_with(
+        ~ paste0("chi"),
+        tidyselect::contains("UPI", ignore.case = FALSE)
+      ) %>%
+      slfhelper::get_anon_chi()
+  }
+  write_file(data, new_file)
+  cli::cli_alert_info("\n {basename(csv_file)} finished at {Sys.time()}")
+  # return(NULL)
+}
```

Comment:
```suggestion
  if (is_chi_in_file(csv_file)) {
    data <- data %>%
      dplyr::rename_with(
        ~ paste0("chi"),
        tidyselect::contains("UPI", ignore.case = FALSE)
      ) %>%
      slfhelper::get_anon_chi()
  } else {
  write_file(data, new_file)
  cli::cli_alert_info("\n {basename(csv_file)} finished at {Sys.time()}")
}
```
Missing a curly bracket here? Do we also want to return NULL?

Reply:
It basically converts chi to anon_chi. If there is no chi, we still write out data to parquet. So no need for `else` part.